test/e2e/atlas/identity_providers_test.go

```diff
@@ -0,0 +1,301 @@
+		cmd := exec.Command(cliPath,
+			federatedAuthenticationEntity,
+			federationSettingsEntity,
+			"describe",
+			"--orgId",
+			orgID,
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.NoError(err, string(resp))
+
+		var settings atlasv2.OrgFederationSettings
+		req.NoError(json.Unmarshal(resp, &settings))
+
+		a := assert.New(t)
+		a.NotEmpty(settings.Id)
+		a.NotEmpty(settings.IdentityProviderStatus)
+		federationSettingsID = settings.Id
+	})
+
+	t.Run("List OIDC IdPs WORKFORCE", func(_ *testing.T) {
+		cmd := exec.Command(cliPath,
+			federatedAuthenticationEntity,
+			identityProviderEntity,
+			"list",
+			"--federationSettingsId",
+			*federationSettingsID,
+			"--protocol",
+			"OIDC",
+			"--idpType",
+			"WORKFORCE",
+			"-o=json",
+		)
+
+		fmt.Println("Printing federationSettingsID")
+		fmt.Println(*federationSettingsID)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.NoError(err, string(resp))
+
+		var settings atlasv2.FederationIdentityProvider
+		req.NoError(json.Unmarshal(resp, &settings))
+	})
+
+	t.Run("List OIDC IdPs WORKLOAD", func(_ *testing.T) {
+		cmd := exec.Command(cliPath,
+			federatedAuthenticationEntity,
+			identityProviderEntity,
+			"list",
+			"--federationSettingsId",
+			*federationSettingsID,
+			"--protocol",
+			"OIDC",
+			"--idpType",
+			"WORKLOAD",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.NoError(err, string(resp))
+
+		var settings atlasv2.FederationIdentityProvider
+		req.NoError(json.Unmarshal(resp, &settings))
+	})
+
+	t.Run("List SAML IdPs", func(_ *testing.T) {
+		cmd := exec.Command(cliPath,
+			federatedAuthenticationEntity,
+			identityProviderEntity,
+			"list",
+			"--federationSettingsId",
+			*federationSettingsID,
+			"--protocol",
+			"SAML",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.NoError(err, string(resp))
+
+		var settings atlasv2.FederationIdentityProvider
+		req.NoError(json.Unmarshal(resp, &settings))
+	})
+
+	t.Run("Create OIDC IdP WORKLOAD", func(t *testing.T) {
+		idpName, err := RandIdentityProviderName()
+		fmt.Println(idpName)
+		req.NoError(err)
+
+		cmd := exec.Command(cliPath,
+			federatedAuthenticationEntity,
+			identityProviderEntity,
+			"create",
+			"oidc",
+			idpName,
+			"--federationSettingsId",
+			*federationSettingsID,
+			"--audience",
+			"AtlasCLIAudience",
+			"--authorizationType",
+			"GROUP",
+			"--desc",
+			"CLI TEST Provider",
+			"--groupsClaim",
+			"groups",
+			"--idpType",
+			"WORKLOAD",
+			"--issuerUri",
+			"https://accounts.google.com",
+			"--userClaim",
+			"user",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.NoError(err, string(resp))
+
+		var provider atlasv2.FederationIdentityProvider
+		req.NoError(json.Unmarshal(resp, &provider))
+
+		a := assert.New(t)
+		a.NotEmpty(provider.Id)
```

Comment:
the use of a variable is only when doing multiple assertions 
```suggestion
		assert.NotEmpty(t, provider.Id)
```
https://github.com/stretchr/testify/?tab=readme-ov-file#assert-package